scripts/bumpversion.py

```diff
@@ -1,48 +1,88 @@
 #!/usr/bin/env python3
 
 import argparse
-import subprocess
+import tempfile
+import textwrap
+import os
+from string import Template
+from subprocess import call
 
-def extract(git_version):
-    version_array = git_version.split('.')
-    patch_array = version_array[2].split('-')
-    major = int(version_array[0][1:])
-    minor = int(version_array[1])
-    patch = int(patch_array[0])
-    postfix = ""
+VERSION_FILE = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../src/version.inc")
+VERSION_CONTENT = """
+#define LC0_VERSION_MAJOR $major 
+#define LC0_VERSION_MINOR $minor
+#define LC0_VERSION_PATCH $patch
+#define LC0_VERSION_POSTFIX $postfix
+"""
+VERSION_CONTENT = textwrap.dedent(VERSION_CONTENT).strip()
+EDITOR = os.environ.get("EDITOR", "nano")
 
-    if len(patch_array) == 3:
-        postfix = patch_array[2]
+
+def get_version():
+    with open(VERSION_FILE, 'r') as f:
+        major = int(f.readline().split()[2])
+        minor = int(f.readline().split()[2])
+        patch = int(f.readline().split()[2])
+        postfix = f.readline().split()[2]
 
     return major, minor, patch, postfix
 
 
+def set_version(major, minor, patch, postfix="\"\""):
+    tmp = Template(VERSION_CONTENT)
+    version_inc = tmp.substitute(major=major, minor=minor, patch=patch, postfix=postfix)
+
+    with open(VERSION_FILE, 'w') as f:
+        f.write(version_inc)
+
+
+def commit(msg):
+    call(["git", "commit", "-am", msg])
+
+
 def tag(major, minor, patch):
     version = "v{}.{}.{}".format(major, minor, patch)
-    subprocess.getoutput("git tag -m {} {}".format(version, version))
+    call(["git", "-m", version, version])
+
+
+def editor(init_msg):
```

Comment:
This seems like deadcode - I can see its potential use in future, but it doesn't look terribly complicated so maybe just drop for now?